modules/sources/enrichers/swaps-usd.ts

```diff
@@ -0,0 +1,69 @@
+import _ from 'lodash';
+import { roundToHour, roundToMidnight } from '../../common/time';
+import { Chain } from '@prisma/client';
+import { prisma } from '../../../prisma/prisma-client';
+import { SwapEvent } from '../../../prisma/prisma-types';
+
+/**
+ * Takes swaps events and enriches them with USD values
+ *
+ * @param swaps
+ * @param chain
+ * @returns
+ */
+export async function swapsUsd(swaps: SwapEvent[], chain: Chain): Promise<SwapEvent[]> {
+    // Enrich with USD values
+    // Group swaps based on timestamp, hourly and daily buckets
+    const groupedSwaps = _.groupBy(swaps, (swap) => {
+        const timestamp = swap.blockTimestamp;
+        // If swap is older than 30 days, round to midnight
+        if (timestamp < Math.floor(Date.now() / 1000) - 30 * 24 * 60 * 60) {
+            return roundToMidnight(timestamp);
+        }
+        // Otherwise round to the nearest hour
+        return roundToHour(timestamp);
+    });
+
+    const dbEntries: SwapEvent[] = [];
+    for (const [timestamp, swaps] of Object.entries(groupedSwaps)) {
+        const tokenPrices = await prisma.prismaTokenPrice.findMany({
+            where: {
+                timestamp: {
+                    gte: parseInt(timestamp),
```

Comment:
shouldnt this be equal to? Otherwise you get a lot of prices and also wont work properly for "refilling" old swaps.

Reply:
yes, good catch.